ddtrace/tracer/span.go

```diff
@@ -379,22 +384,24 @@ func (s *span) SetOperationName(operationName string) {
 	// We don't lock spans when flushing, so we could have a data race when
 	// modifying a span as it's being flushed. This protects us against that
 	// race, since spans are marked `finished` before we flush them.
-	if s.finished {
+	if s.finished && !s.inProcessor {
 		// already finished
 		return
 	}
 	s.Name = operationName
 }
 
-func (s *span) finish(finishTime int64) {
+// finish reports whether there are more spans left in the trace
+// which s belongs to.
+func (s *span) finish(finishTime int64) bool {
 	s.Lock()
 	defer s.Unlock()
 	// We don't lock spans when flushing, so we could have a data race when
 	// modifying a span as it's being flushed. This protects us against that
 	// race, since spans are marked `finished` before we flush them.
 	if s.finished {
 		// already finished
-		return
+		return true
```

Comment:
```suggestion
		return false
```

Reply:
This isn't a problem anymore now that the processor was moved to the writer, marking as resolved.